src/site/content/en/blog/virtualkeyboard/index.md

```diff
@@ -0,0 +1,208 @@
+
+### Opting in to the new virtual keyboard behavior
+
+To tell the browser that you are taking care of virtual keyboard occlusions yourself, you need to
+first opt in to the new virtual keyboard behavior by setting the boolean property `overlaysContent`
+to `true`.
+
+```js
+navigator.virtualKeyboard.overlaysContent = true;
+```
+
+### Showing and hiding the virtual keyboard
+
+You can programmatically show the virtual keyboard by calling its `show()` method. For this to work,
+the focused element needs to be a form control (such as a `textarea` element), or be an editing host
+(for example, by using the
+[`contenteditable`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/contentEditable)
+attribute). The method always returns `undefined` but triggers a `geometrychanged` event
+if the virtual keyboard previously was not shown.
+
+```js
+navigator.virtualKeyboard.show();
+```
+
+To hide the virtual keyboard, call the `hide()` method. The method always returns `undefined` but triggers
+a `geometrychanged` event if the virtual keyboard previously was shown.
+
+```js
+navigator.virtualKeyboard.hide();
+```
+
+### Being informed of geometry changes
+
+Whenever the virtual keyboard appears or disappears, the `geometrychanged` event is dispatched. The
+event's `target` property contains the new geometry of the virtual keyboard inset as a
+[`DOMRect`](https://www.w3.org/TR/geometry-1/#domrect).
+The inset corresponds to the top, right, bottom, and/or left properties.
+
+```js
+navigator.virtualKeyboard.addEventListener('geometrychanged', (event) => {
+  const { x, y, width, height } = event.target;
+  console.log('Virtual keyboard geometry changed:', x, y, width, height);
+});
+```
+
+### Getting the current geometry
+
+You can get the current geometry of the virtual keyboard by looking at the `boundingRect` property.
+It exposes the current dimensions of the virtual keyboard as a
+[`DOMRect`](https://www.w3.org/TR/geometry-1/#domrect) object.
+
+```js
+const { x, y, width, height } = navigator.virtualKeyboard.boundingRect;
+console.log('Virtual keyboard geometry:', x, y, width, height);
+```
+
+### The CSS environment variables
+
+The VirtualKeyboard API exposes a set of CSS environment variables that provide information about
+the virtual keyboard's appearance.
+They are modeled similar to the [`inset`](https://developer.mozilla.org/en-US/docs/Web/CSS/inset) CSS property,
+that is, corresponding to the top, right, bottom, and/or left properties.
+
+- `keyboard-inset-top`
+- `keyboard-inset-right`
+- `keyboard-inset-bottom`
+- `keyboard-inset-left`
+- `keyboard-inset-width`
+- `keyboard-inset-height`
+
+The virtual keyboard insets are six environment variables that define a rectangle by its top, right,
+bottom, and left insets from the edge of the viewport. The width and height insets are calculated
+from the remaining insets for developer ergonomics. The default value of each keyboard inset is
```

Comment:
```suggestion
from the other insets for developer ergonomics. The default value of each keyboard inset is
```

Reply:
This takes some explanation. You have two groups: width and height; everything else, both referenced in this sentence. When my Anglophone ears here remaining, I reread the preceding text looking for a third group that I don't remember reading about.

Reply:
Also, does this mean that width and height are read only?

Reply:
Everything is read-only. You can't change the virtual keyboard's size.